camelot/view/action_steps/application.py

```diff
@@ -66,16 +66,25 @@ def render( self, gui_context ):
         main_window_context = gui_context.copy()
         main_window_context.progress_dialog = None
         main_window_context.admin = self.admin
-        main_window = MainWindow( gui_context=main_window_context )
+
+        # Check if a QMainWindow already exists
+        window = None
+        app = QtWidgets.QApplication.instance()
+        for widget in app.allWidgets():
+            if widget.__class__.__name__ == 'QMainWindow':
```

Comment:
gebruik hier aub 'isinstance', dan w subclasses ook gedetecteerd

Reply:
Dit is inderdaad beter.